core/src/main/java/org/infinispan/util/CoreBlockHoundIntegration.java

```diff
@@ -74,9 +85,6 @@ private static void registerBlockingMethods(BlockHound.Builder builder) {
    }
 
    private static void questionableMethodsAllowedToBlock(BlockHound.Builder builder) {
-      // Component registry has a lock to protect its state - is short lived lock
-      builder.allowBlockingCallsInside(BasicComponentRegistryImpl.class.getName(), "prepareWrapperChange");
-
       // This one should probably not be allowed - it is waiting for another component to start
       // TODO: This might actually be a bug in the rewiring logic. (shows in StateTransferOverwritingValueTest)
       builder.allowBlockingCallsInside(BasicComponentRegistryImpl.class.getName(), "awaitWrapperState");
```

Comment:
Can you check this again? I commented it out locally and I didn't get any failures.

Reply:
Possibly, this is one from over a month ago.

Reply:
Yeah, I tested when I was reviewing your first BlockHound PR but I forgot to mention it.

Reply:
> https://ci.infinispan.org/job/Infinispan/job/PR-7952/3/testReport/junit/org.infinispan.distribution.rehash/StateTransferOverwriteTest%5BDIST_SYNC,%20tx=false%5D/Blocking/

Now that I see the stack trace, it's "normal". `ClusterTopologyManagerImpl` uses `new ActionSequencer(nonBlockingExecutor, true)`, which forces the view handling to be executed on the non-blocking executor even when it doesn't have to block. During startup, normally the components start in a single thread, so there is no call to `awaitWrapperState`. But since CTMI handles the initial view in a separate thread, the main thread and the view handling thread could be trying to start `LocalTopologyManagerImpl` at the same time, and one of them will have to wait.

There are 3 options:
1. Keep the exception for `awaitWrapperState`, but remove the TODO and say instead it's ok because it only happens during startup
2. Create the `ActionSequencer` with `forceExecutor = false`, and explicitly submit to the non-blocking thread pool in `ClusterViewListener#handleViewChange`.
    We need to switch threads there because JGroups' GMS will only mcast the view to other members after our listener returns.
3. Create the `ActionSequencer` on the blocking executor, "just in case".
    Since I've removed almost all blocking, and some tasks like computing CHs use lots of CPU, I'd rather not do this.

Reply:
I created https://issues.redhat.com/browse/ISPN-11396 to look into this further. For now I will just comment on it and mention the JIRA.